### ISSUE TITLE
Incorrect prefer_is_empty warning

### ISSUE THREAD

**Issue by spkersten** [dart-lang/linter#2151]
Consider:

```dart
class Foo {
  const Foo(List<int> numbers)
      : assert(numbers.length > 0),
        assert(numbers.isNotEmpty);
}
```

The first assert results in a warning say:
```
info: Use isNotEmpty instead of length (prefer_is_empty at [...] ...)
```

But following that suggestion (the second assert) is an error:
```
error: Invalid constant value. (invalid_constant at [...] ...)
```

Dart 2.7.0

**Comment by srawlins** [comment]
Thanks for opening!

**Comment by bruno-garcia** [comment]
This came up in #1719
I believe it was fixed through #2122 and landed in master June 2nd and is on the release notes of `0.1.117`

I'm on `Dart version 2.9.0 (build 2.9.0-21.10.beta)` and still affected by this issue. 

Not sure how the linter release gets through into Dart to check versions
